src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,128 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import classNames from 'classnames';
 import React from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  todos: Todo[],
+  isLoading: boolean,
+  selectedUserId: number,
+  setTodosByChecked: (id: number) => void;
+  setSelectedUserId: (id: number) => void;
+};
+
+type State = {
+  searchRequest: string,
+  filterBy: string,
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    searchRequest: '',
+    filterBy: 'all',
+  };
+
+  setChangedValue = (e: React.ChangeEvent<HTMLSelectElement>
+  | React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Уникай однобуквенних назв змінних
```suggestion
  setChangedValue = (
  event: React.ChangeEvent<
      HTMLSelectElement | HTMLInputElement
    >
  ) => {
```